internal/statusapi/statusapi.go

```diff
@@ -0,0 +1,107 @@
+package statusapi
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/fabric8-services/fabric8-jenkins-proxy/internal/clients"
+	"github.com/fabric8-services/fabric8-jenkins-proxy/internal/configuration"
+	"github.com/fabric8-services/fabric8-jenkins-proxy/internal/proxy"
+	"github.com/julienschmidt/httprouter"
+	log "github.com/sirupsen/logrus"
+)
+
+//JenkinsStatusAPI contains API to check whether Jenkins for the current user is idle|running|starting
+type JenkinsStatusAPI interface {
+	Status(w http.ResponseWriter, r *http.Request, ps httprouter.Params)
+}
+
+// JenkinsStatusAPIImpl implements JenkinsStatusAPI
+type jenkinsStatusAPIImpl struct {
+	tenant *clients.Tenant
+	idler  clients.IdlerService
+}
+
+// JenkinsStatus represents whether jenkins is running|starting|idle
+type JenkinsStatus struct {
+	Status string `json:"status"`
+}
+
+// NewJenkinsStatusAPI creates a new instance of JenkinsStatusAPI
+func NewJenkinsStatusAPI(tenant *clients.Tenant, idler clients.IdlerService) JenkinsStatusAPI {
```

Comment:
shouldn't the `idler` be a pointer as well?

Reply:
in that case, shouldn't it be tenant interface?

Reply:
there isnt one